utt/utt-client-api/src/User.cpp

```diff
@@ -13,85 +13,430 @@
 
 #include "utt-client-api/User.hpp"
 
+// libutt
+#include <utt/Params.h>
+#include <utt/RegAuth.h>
+#include <utt/RandSig.h>
+#include <utt/Address.h>
+#include <utt/Coin.h>
+#include <utt/MintOp.h>
+#include <utt/BurnOp.h>
+#include <utt/DataUtils.hpp>
+
+// libutt new interface
+#include <UTTParams.hpp>
+#include <coin.hpp>
+#include <client.hpp>
+#include <config.hpp>
+#include <budget.hpp>
+#include <mint.hpp>
+#include <burn.hpp>
+#include <transaction.hpp>
+#include <serialization.hpp>
+
+#include "utt-client-api/PickCoins.hpp"
+
 namespace utt::client {
 
-std::vector<uint8_t> User::getRegistrationInput() const {
-  // [TODO-UTT] Implement User::getRegistrationInput
-  return std::vector<uint8_t>{};
+struct User::Impl {
+  utt::Transaction createTx_Burn(const libutt::api::Coin& coin);
+
+  utt::Transaction createTx_Self1t2(const libutt::api::Coin& coin, uint64_t amount);
+  utt::Transaction createTx_Self2t1(const std::vector<libutt::api::Coin>& coins);
+  utt::Transaction createTx_Self2t2(const std::vector<libutt::api::Coin>& coins, uint64_t amount);
+
+  utt::Transaction createTx_1t1(const libutt::api::Coin& coin, const std::string& userId, const std::string& pk);
+  utt::Transaction createTx_1t2(const libutt::api::Coin& coin,
+                                uint64_t amount,
+                                const std::string& userId,
+                                const std::string& pk);
+  utt::Transaction createTx_2t1(const std::vector<libutt::api::Coin>& coins,
+                                const std::string& userId,
+                                const std::string& pk);
+  utt::Transaction createTx_2t2(const std::vector<libutt::api::Coin>& coins,
```

Comment:
Why not `1to2`, `2to1` etc. instead of `1t2`. `2t1`, ...? Unless I'm missing something and the "t" means something else?